benchmark_runner/common/elasticsearch/elasticsearch_operations.py

```diff
@@ -201,10 +198,99 @@ def update_elasticsearch_index(self, index: str, id: str, metadata: dict = ''):
     @logger_time_stamp
     def get_elasticsearch_index_by_id(self, index: str, id: str):
         """
-        This method return elastic search index data by id
+        This method return elasticsearch index data by id
         :param index: index name
         :param id: The specific index id
         :return:
         """
         return self.__es.get(index=index, id=id)
 
+    @typechecked()
+    @logger_time_stamp
+    def delete_elasticsearch_index_by_id(self, index: str, id: str):
```

Comment:
You probably want to check for either index or id being None and raise an exception (that's almost certainly not something you want).

For convenience, you might want to allow id to be either a string or a list of strings, and do the iteration inside this function.

Reply:
I will add check for input.
Id should be string, who need a list can wrap it simply by for loop